Add debug capabilities to orawls

BasLangenberg opened the pull request:
This commit adds the use of a temp directory to orawls. When you create
the following line in wls_settings on the host the module is ran,
orawls will copy all the scripts created as tempfiles to
/tmp/orawls-archive. This is useful when new or modified types are debugged.

debug_module: "true"

wls_settings needs to be changed, so we can puppetize this value as
well.

BasLangenberg commented:
The code should be pretty self explanatory. 

I would like to  see this setting to be controlled by Puppet, but I need to do some more research on how this is exactly implemented right now. Consider not merging, so we can add this in one go.

BasLangenberg commented:
Hi Edwin,

Should be done now. Can you please review and let me know what you think.

Cheers!

Bas

biemond commented:
nice,

when you don't provide debugmode does it crash on this 'debug_module.to_s' and maybe I am missing the debug parameter entry inside wls_setting

also add tmp_dir parameter to wls_setting with /tmp as default and use that one.

BasLangenberg commented:
I'll update the PR later today.

Regarding the tmp_dir parameter, it might be better to add an archive_dir parameter so people can specify the full path on where the files are stored. This will default to /tmp/orawls-archive. What do you think?

BasLangenberg commented:
It should not crash if you do not set debug mode. The parameter has been configured to set it to false automatically if it's not provided. I just tested this so it should work in all cases.

I'm currently working on adding the parameter for the archive directory, as I think we should indeed make that configurable.

BasLangenberg commented:
I have added the parameter and fixed the Travis build too. I forgot to add a file in my previous commit. Apologies for that.

The code should be ok now.

biemond commented:
:+1:
